git-traverse/src/commit.rs

```diff
@@ -151,6 +177,80 @@ pub mod ancestors {
         type Item = Result<ObjectId, Error>;
 
         fn next(&mut self) -> Option<Self::Item> {
+            match self.sorting {
+                Sorting::GraphOrder => self.graph_sort_next(),
+                Sorting::ByCommitterDate => self.next_by_commit_date(),
+            }
+        }
+    }
+
+    impl<Find, Predicate, StateMut> Ancestors<Find, Predicate, StateMut>
+    where
+        Find: for<'a> FnMut(&oid, &'a mut Vec<u8>) -> Option<CommitRefIter<'a>>,
+        Predicate: FnMut(&oid) -> bool,
+        StateMut: BorrowMut<State>,
+    {
+        fn next_by_commit_date(&mut self) -> Option<Result<ObjectId, Error>> {
+            let state = self.state.borrow_mut();
+            state.parents_with_date.clear();
+            let res = state.next.pop_front();
+
+            if let Some(oid) = res {
+                match (self.find)(&oid, &mut state.buf) {
+                    Some(mut commit_iter) => {
+                        if let Some(Err(decode_tree_err)) = commit_iter.next() {
+                            return Some(Err(decode_tree_err.into()));
+                        }
+
+                        for token in commit_iter {
+                            match token {
+                                Ok(git_object::commit::ref_iter::Token::Parent { id }) => {
+                                    let mut vec = vec![];
+                                    let parent = (self.find)(id.as_ref(), &mut vec);
+
+                                    let parent_committer_date = parent
+                                        .and_then(|mut parent| parent.committer().map(|committer| committer.time));
+
+                                    if let Some(parent_committer_date) = parent_committer_date {
+                                        state.parents_with_date.push((id, parent_committer_date.time));
+                                    }
+
+                                    if matches!(self.mode, Parents::First) {
+                                        break;
+                                    }
+                                }
+                                Ok(_unused_token) => break,
+                                Err(err) => return Some(Err(err.into())),
+                            }
+                        }
+                    }
+                    None => return Some(Err(Error::NotFound { oid })),
+                }
+            }
+
+            state
+                .parents_with_date
+                .sort_by(|(_, time), (_, other_time)| other_time.cmp(&time));
```

Comment:
Here I would find `time.cmp(&other_time).reverse()` clearer, the `reverse()` makes a bit more obvious that this is going to be descending, as opposed to comparing the arguments in another order.